src/common/getServiceKeyIndex.ts

```diff
@@ -0,0 +1,5 @@
+export function getServiceKeyIndex(url: string): string | undefined {
+  try {
+    return new URL(url).host
+  } catch {}
+}
```

Comment:
`.host` is never `undefined` so omit it in the return type.

I see that you're using `undefined` as a case at the call-site. It seems like what you want is a utility that does a lookup and returns a string or undefined if the lookup succeeds or fails. I made a fixup commit to `aj/add-service-keys-for-evm-currencies` branch which you can pull from that has the suggested solution.